nessus_ec2.tf

```diff
@@ -33,6 +33,15 @@ resource "aws_instance" "nessus" {
   count    = lookup(var.operations_instance_counts, "nessus", 0)
   provider = aws.provisionassessment
 
+  # When a Nessus instance starts up, it executes cloud-init/nessus-setup.sh,
+  # which requires access to the STS and SSM endpoints.  To ensure that access
+  # is available, we force dependencies on the security group rules that
+  # allow STS and SSM endpoint access from Nessus.
+  depends_on = [
+    aws_security_group_rule.ingress_from_nessus_to_sts_via_https,
+    aws_security_group_rule.ingress_from_nessus_to_ssm_via_https
```

Comment:
You may want to list the VPC endpoints themselves as well.  They are secretly EC2 instances under the hood, and they take some time to create.

Sorry I keep having these thoughts _after_ the PR has been approved.

Reply:
Just to be clear, are you talking about adding dependencies on `aws_vpc_endpoint.sts` and `aws_vpc_endpoint.ssm` here?

https://github.com/cisagov/cool-assessment-terraform/blob/1896e889886d71aaf0f1c834935fa927fac201c6/vpc_endpoints.tf#L9-L35

Also, do I need to include `aws_vpc_endpoint.ssmmessages` also?

https://github.com/cisagov/cool-assessment-terraform/blob/1896e889886d71aaf0f1c834935fa927fac201c6/vpc_endpoints.tf#L76-L87

Reply:
Yes, I'm talking about adding dependencies on `aws_vpc_endpoint.sts` and `aws_vpc_endpoint.ssm`. 
Does cloud-init need access to any services after SSM Parameter Store, such S3?  If so then those should be listed as well.

`aws_vpc_endpoint.ec2`, `aws_vpc_endpoint.ec2messages`, `aws_vpc_endpoint.kms`, and `aws_vpc_endpoint.ssmmessages` are all used by the Amazon AWS SSM agent, but should not be necessary for accessing SSM Parameter Store; hence, there is no need to list those endpoints in the `depends_on`.

Reply:
The only cloud-init in this repo that needs S3 is for the Guacamole instance to fetch it's SSL certificate (see [install-certificates.py](https://github.com/cisagov/cool-assessment-terraform/blob/develop/cloud-init/install-certificates.py)).  I can add a `depends_on` to the Guacamole instance for `aws_vpc_endpoint.s3` as well.

Reply:
This has been taken care of in #103 - thanks @jsf9k for the suggestions!